src/relay/transforms/type_infer.cc

```diff
@@ -98,11 +100,11 @@ class TypeInferencer : private ExprFunctor<Type(const Expr&)>,
  public:
   // constructors
 
-  explicit TypeInferencer(IRModule mod, GlobalVar current_func)
+  explicit TypeInferencer(IRModule mod, GlobalVar current_func, DiagnosticContext diag_ctx)
       : mod_(mod),
         current_func_(current_func),
-        err_reporter(),
-        solver_(current_func, mod, &this->err_reporter) {
+        diag_ctx(diag_ctx),
+        solver_(current_func, mod, nullptr) {
```

Comment:
use optional

Reply:
This is going to go away, I'm like 50% of the way through fixing the type checker. A lot of things have changed since we have designed it.